src/hooks/react/useDebounceState.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
```

Comment:
What do you think about this implementation @tomusdrw @wirednkod ? Do you get the idea of this hook?